### ISSUE TITLE
Opaque Pointers

### ISSUE THREAD

**Issue by ChocolateLoverRaj** [f0rki/mapping-high-level-constructs-to-llvm-ir#49]
LLVM pointers don't include the type of what it's pointing to anymore - https://llvm.org/docs/OpaquePointers.html, so basically `i8*` becomes `ptr`. Examples in this repo (idk if all or just some) use the old pointer style.

**Comment by f0rki** [comment]
Good point. Most of the LLVM examples are already outdated and I hardly find the time to update it. I'll gladly accept any help. I think llvm is capable of reading old bitcode so we be able to update the samples by essentially piping through opt?
